test-suite/shared/src/test/require-jdk11/org/scalajs/testsuite/javalib/io/InputStreamTestOnJDK11.scala

```diff
@@ -39,6 +39,28 @@ class InputStreamTestOnJDK11 {
   private def assertBytesEqual(expect: Seq[Int], got: Array[Byte]) =
     assertArrayEquals(expect.toArray.map(_.toByte), got)
 
+  @Test def readAllBytes(): Unit = {
+    assertBytesEqual(0 until 100, chunkedStream(10, 0 until 100).readAllBytes())
+    assertBytesEqual(Nil, emptyStream().readAllBytes())
+  }
+
+  @Test def readNBytes(): Unit = {
+    assertBytesEqual(0 until 20, chunkedStream(10, 0 until 100).readNBytes(20))
+    assertBytesEqual(0 until 100, chunkedStream(10, 0 until 100).readNBytes(200))
+    assertBytesEqual(Nil, chunkedStream(10, 0 until 100).readNBytes(0))
+    assertBytesEqual(Nil, emptyStream().readNBytes(1000))
```

Comment:
AFAICT, those tests don't exercise the code path to resize `buf` when reading more than 1024 bytes.

Reply:
Good catch (and of course it discovered bugs).

Note that while fixing them, I have also limited the buffer size doubling to `len`.